src/Microsoft.AspNetCore.Blazor/Routing/Router.cs

```diff
@@ -74,15 +79,21 @@ protected virtual void Render(RenderTreeBuilder builder, Type handler, IDictiona
             builder.CloseComponent();
         }
 
-        private void Refresh()
+        private void Refresh(bool useFallback = false)
         {
-            var locationPath = UriHelper.ToBaseRelativePath(_baseUri, _locationAbsolute);
+            var locationPath = useFallback ? FallbackRoute : UriHelper.ToBaseRelativePath(_baseUri, _locationAbsolute);
             locationPath = StringUntilAny(locationPath, _queryOrHashStartChar);
             var context = new RouteContext(locationPath);
             Routes.Route(context);
             if (context.Handler == null)
             {
-                throw new InvalidOperationException($"'{nameof(Router)}' cannot find any component with a route for '/{locationPath}'.");
+                if (!useFallback && FallbackRoute != null)
+                {
+                    Refresh(useFallback: true);
+                    return;
+                }
+                else
+                    throw new InvalidOperationException($"'{nameof(Router)}' cannot find any component with {(useFallback ? "the fallback route" : "a route for")} '/{locationPath}'.");
```

Comment:
This can be simplified to
```csharp
if (useFallback || FallbackRoute == null)
    throw new InvalidOperationException($"'{nameof(Router)}' cannot find any component with {(useFallback ? "the fallback route" : "a route for")} '/{locationPath}'.");

Refresh(useFallback: true);
```

Reply:
Thanks! Will change it.

Reply:
Actually, I need the return in there otherwise it will fail because the original route gets rendered anyways in certain cases.

Reply:
Oh good point, my bad, i missed that the method didn't end there

Reply:
Fixed.